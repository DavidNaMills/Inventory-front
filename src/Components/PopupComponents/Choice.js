import React from 'react';

const Choice = (props) =>{
    const {message, onCancel, onAccept} = props;

    return (
        <div>
            <p>{message}</p>
            <button onClick={onAccept}>Accept</button>
            <button onClick={onCancel}>Cancel</button>
        </div>
    )
}

export default Choice;