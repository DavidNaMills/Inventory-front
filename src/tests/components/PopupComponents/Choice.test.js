import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Choice from '../../../Components/PopupComponents/Choice';

const mockProps = {
    message: 'are you sure?',
    onAccept: jest.fn(),
    onCancel: jest.fn(),
}

afterEach(()=>{
    jest.clearAllMocks();
})

describe('<Choice/> popup component test suite', ()=>{
    const wrapper = shallow(<Choice {...mockProps} />);

    it('matches snapshot', ()=>{
        const comp = renderer.create(<Choice {...mockProps} />);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });
    
    it('exists', ()=>{
        expect(wrapper.exists).toBeTruthy();
    });
    
    it('renders a <div>, a <p> with the message, and 2 <button> elements', ()=>{
        expect(wrapper.find('div').length).toBe(1);
        expect(wrapper.find('p').length).toBe(1);
        expect(wrapper.find('p').text()).toEqual(mockProps.message);
    });
    
    it('fires the onAccept method when clicked', ()=>{
        wrapper.find('button').at(0).simulate('click');
        expect(mockProps.onAccept).toHaveBeenCalledTimes(1);
    });
    
    it('fires the onCancel method when clicked', ()=>{
        wrapper.find('button').at(1).simulate('click');
        expect(mockProps.onCancel).toHaveBeenCalledTimes(1);

    });
});