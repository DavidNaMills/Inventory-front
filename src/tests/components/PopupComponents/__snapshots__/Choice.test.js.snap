// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Choice/> popup component test suite matches snapshot 1`] = `
<div>
  <p>
    are you sure?
  </p>
  <button
    onClick={[MockFunction]}
  >
    Accept
  </button>
  <button
    onClick={[MockFunction]}
  >
    Cancel
  </button>
</div>
`;
